### PR TITLE
Add the possibility to use an attributedPlaceholder

### PR DESCRIPTION
**Pull request by pblondin**
Feel free to merge if you want. I added to the possibility to use an attributedPlaceholder. Also defaultPlaceholderColor function should not use the private API.

### CODE REVIEW

**Comment by devxoul** [comment]
Hi @pblondin,
I'll check this soon.
Thank you for your time!

**Comment by devxoul** [comment]
I intended placeholder to behave as that of UILabel. UILabel doesn't have `attributedPlaceholder` currently, so it is not time to be merged.

Thank you again for your time!

**Comment by TomLiu** [comment]
I think placeholder should behave as UITextField not UILabel and UITextField do has an `attributedPlaceholder `

**Comment by devxoul** [comment]
Oh, god. I meant `UITextField`, not `UILabel`. My mistake.
I'll check this soon!

File: UITextView+Placeholder/UITextView+Placeholder.m [hunk 1]
```diff
@@ -55,14 +55,7 @@ - (void)swizzledDealloc {
...
 + (UIColor *)defaultPlaceholderColor {
-    static UIColor *color = nil;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        UITextField *textField = [[UITextField alloc] init];
-        textField.placeholder = @" ";
-        color = [textField valueForKeyPath:@"_placeholderLabel.textColor"];
-    });
-    return color;
+    return [UIColor colorWithWhite:0.70f alpha:1.0f];
```
Comment (devxoul) [inline UITextView+Placeholder/UITextView+Placeholder.m]: Placeholder color has an alpha value. Check with colored background.

Comment (pblondin) [inline UITextView+Placeholder/UITextView+Placeholder.m]: I'm not sure the follow you. The default color for placeholder (as stated [here](https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UITextField_Class/index.html#//apple_ref/occ/instp/UITextField/placeholder)) is 70% grey.

I do agree that your method is more practical, as Apple may change the default color of placeholder at any time, but your are using the private API to obtain the color which is not allowed.

Comment (devxoul) [inline UITextView+Placeholder/UITextView+Placeholder.m]: **Reason 1.**
This method is allowed by Apple. It's just private property, not a private API. All of my applications using this are uploaded to AppStore.

or you can refer this StackOverflow answers: http://stackoverflow.com/questions/1340224/iphone-uitextfield-change-placeholder-text-color

**Reason 2**
Apple is lying. Placeholder label color isn't 70% grey.

```swift
let textField = UITextField()
textField.placeholder = "A"
if let color = textField.valueForKeyPath("_placeholderLabel.textColor") as? UIColor {
    println(color) // UIDeviceRGBColorSpace 0 0 0.0980392 0.22
}
```

It's 22% of very dark blue.

Comment (devxoul) [inline UITextView+Placeholder/UITextView+Placeholder.m]: @pblondin ping :sparkles:

**Comment by devxoul** [comment]
I've merged only `attributedPlaceholder` lines: https://github.com/devxoul/UITextView-Placeholder/commit/5cdda39a818195597cac627a1dcfbd03e028b7c0

**Pull request closed without merging**